test/contracts/pass/dsProvePass.sol

```diff
@@ -48,29 +48,22 @@ contract SolidityTest is DSTest {
         assertEq(b, c.a());
     }
 
-    function proveFail_revertSmoke() public {
+    function prove_no_fail_require() public {
         require(false);
     }
 
+    function proveFail_userAssertSmoke() public {
+        assert(false);
+    }
+
     function proveFail_assertSmoke() public {
         assertTrue(false);
     }
 
-    // Takes too long, disabling here, moving to benchmarks
-    // function prove_transfer(uint supply, address usr, uint amt) public {
-    //     if (amt > supply) return; // no underflow
-    //
-    //     token.mint(address(this), supply);
-    //
-    //     uint prebal = token.balanceOf(usr);
-    //     token.transfer(usr, amt);
-    //     uint postbal = token.balanceOf(usr);
-    //
-    //     uint expected = usr == address(this)
-    //                     ? 0    // self transfer is a noop
-    //                     : amt; // otherwise `amt` has been transfered to `usr`
-    //     assertEq(expected, postbal - prebal);
-    // }
+    // requires don't fail passing tests, but are not treated as successes in proveFail tests.
+    function proveFail_require() public {
```

Comment:
I am confused, sorry. This is supposed to not fail, because it's a `require`. However, this says `proveFail`, so it's expecting a failure. So this test fails. However, this is in the file `dsProvePass.sol`, so this file is supposed to pass, and it can only pass if this failure is proved, but this failure will not happen, due to `require` so this will fail the test. Yet it passes. I'm confused, sorry. Can you explain what's going on here?

Reply:
this test is just wrong. It should be in `dsProveFail`.

The semantics should IMO be:

`prove_`: only reachable user defined assertion violations or ds-test assertion violations should trigger a failure
`proveFail_`: all branches should either revert or trigger a ds-test assertion violation

Reply:
wait no, this test should actually be here. lol. yeah this is bit of a mindfuck, think we can probably just remove the `proveFail` stuff in the long term.